### PR TITLE
Add --from-branch cli arg

### PR DESCRIPTION
**Pull request by MichaelStubbings**
This allows users to choose which skeleton branch they want to merge into their code. If they do not supply this argument, then main will be used by default.

- Add from branch argument
- Add unit tests

### CODE REVIEW

File: src/python3_pip_skeleton/__main__.py [hunk 1]
```diff
@@ -234,6 +242,11 @@ def main(args=None):
     sub.add_argument(
         "--package", default=None, help="Package name, defaults to directory name"
     )
+    sub.add_argument(
+        "--from-branch",
+        default=None,
```
Comment (AlexanderWells-diamond) [inline src/python3_pip_skeleton/__main__.py:247]: Is there a reason you couldn't just write `"main"` here?

Comment (MichaelStubbings) [reply src/python3_pip_skeleton/__main__.py:247]: No, but that would be simpler I agree

Comment (MichaelStubbings) [reply src/python3_pip_skeleton/__main__.py:247]: Oh I remember, it was to keep it consistent with the existing code. Notice how all arguments default to None and get set later in merge_skeleton.

**Review by gilesknap — APPROVED** [review]
nice tests.

**Pull request merged**